src/Features/LanguageServer/Protocol/Extensions/Extensions.cs

```diff
@@ -59,6 +61,23 @@ public static Uri GetUriFromProjectPath(this TextDocument document)
             return ProtocolConversions.GetUriFromFilePath(path);
         }
 
+        public static Uri GetUriFromFolders(this TextDocument document)
+        {
+            Contract.ThrowIfNull(document.Name);
+            Contract.ThrowIfNull(document.Project.FilePath);
+            Contract.ThrowIfFalse(document.Folders.Any());
+
+            var projectDirectoryName = Path.GetDirectoryName(document.Project.FilePath);
+            Contract.ThrowIfNull(projectDirectoryName);
+            using var _ = ArrayBuilder<string>.GetInstance(out var pathBuilder);
```

Comment:
Possibly premature optimization: in this case we'll know the total size the array should be, so you could supply that up-front.